server/backup_runner.go

```diff
@@ -115,78 +115,70 @@ func (s *BackupRunner) Run(ctx context.Context, wg *sync.WaitGroup) {
 	}
 }
 
-func (s *BackupRunner) scheduleBackupTask(ctx context.Context, database *api.Database, backupName string) error {
+func (s *Server) scheduleBackupTask(ctx context.Context, database *api.Database, backupName string, backupType api.BackupType, storageBackend api.BackupStorageBackend, creatorID int) (*api.Backup, error) {
+	backupCreate := &api.BackupCreate{
+		CreatorID:      creatorID,
+		DatabaseID:     database.ID,
+		Name:           backupName,
+		StorageBackend: storageBackend,
+		Type:           backupType,
+	}
 	path := getBackupRelativeFilePath(database.ID, backupName)
-	if err := createBackupDirectory(s.server.profile.DataDir, database.ID); err != nil {
-		return err
+	if err := createBackupDirectory(s.profile.DataDir, database.ID); err != nil {
+		return nil, fmt.Errorf("failed to create backup directory, error: %w", err)
 	}
+	backupCreate.Path = path
```

Comment:
nit: if we move line 126 to be the first in the function, Path can be set as part of &api.BackupCreate{}.